### ISSUE TITLE
tm_rgb() seems to fail for higher resolution rasters?

### ISSUE THREAD

**Issue by philament** [mtennekes/tmap#442]
This works:

```
bg = tmaptools::read_osm(NLD_prov, zoom = 8)
tm_shape(bg) + tm_rgb()
```

![image](https://user-images.githubusercontent.com/31984490/81271561-8fa39c80-904c-11ea-8fd7-f36d0c3dd084.png)

Changing the zoom factor to a higher level results in a messed up plot (tested and reproduced on both Mac OS and Windows):

```
bg = tmaptools::read_osm(NLD_prov, zoom = 9)
tm_shape(bg) + tm_rgb()
```

![image](https://user-images.githubusercontent.com/31984490/81271681-b2ce4c00-904c-11ea-8cfe-ae4d74018991.png)

```
R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] remotes_2.1.1            rnaturalearthhires_0.2.0 RColorBrewer_1.1-2       raster_3.1-5             sp_1.4-1                 forcats_0.5.0           
 [7] stringr_1.4.0            dplyr_0.8.5              purrr_0.3.4              readr_1.3.1              tidyr_1.0.2              tibble_3.0.1            
[13] ggplot2_3.3.0            tidyverse_1.3.0          tmaptools_3.0-1          tmap_3.1                 sf_0.9-2                

loaded via a namespace (and not attached):
 [1] nlme_3.1-147        fs_1.4.1            lubridate_1.7.8     httr_1.4.1          rprojroot_1.3-2     tools_3.6.0         backports_1.1.6     utf8_1.1.4         
 [9] rgdal_1.4-8         R6_2.4.1            KernSmooth_2.23-17  DBI_1.1.0           colorspace_1.4-1    withr_2.2.0         OpenStreetMap_0.3.4 tidyselect_1.0.0   
[17] prettyunits_1.1.1   processx_3.4.2      leaflet_2.0.3       curl_4.3            compiler_3.6.0      leafem_0.1.1        cli_2.0.2           rvest_0.3.5        
[25] xml2_1.3.2          scales_1.1.0        classInt_0.4-3      callr_3.4.3         digest_0.6.25       base64enc_0.1-3     dichromat_2.0-0     pkgconfig_2.0.3    
[33] htmltools_0.4.0     dbplyr_1.4.3        htmlwidgets_1.5.1   rlang_0.4.6         readxl_1.3.1        rstudioapi_0.11     generics_0.0.2      jsonlite_1.6.1     
[41] crosstalk_1.1.0.1   magrittr_1.5        Rcpp_1.0.4.6        munsell_0.5.0       fansi_0.4.1         abind_1.4-5         lifecycle_0.2.0     stringi_1.4.6      
[49] leafsync_0.1.0      pkgbuild_1.0.7      grid_3.6.0          parallel_3.6.0      crayon_1.3.4        lattice_0.20-41     stars_0.4-1         haven_2.2.0        
[57] hms_0.5.3           ps_1.3.2            pillar_1.4.3        codetools_0.2-16    reprex_0.3.0        XML_3.99-0.3        glue_1.4.0          modelr_0.1.7       
[65] png_0.1-7           vctrs_0.2.4         cellranger_1.1.0    gtable_0.3.0        assertthat_0.2.1    lwgeom_0.2-3        broom_0.5.6         e1071_1.7-3        
[73] class_7.3-17        viridisLite_0.3.0   rJava_0.9-12        units_0.6-6         ellipsis_0.3.0 
```

**Comment by mtennekes** [comment]
Works fine under Ubuntu.

Are you sure you have the latest tmap verison from github? You should also see the message `stars object downsampled to 931 by 1074 cells. See tm_shape manual (argument raster.downsample)`.

Does `plot(bg)` work? If not, then there is a problem with OSM (the R package `OpenStreetMap`, the Java installation, or OSM itself).

**Comment by philament** [comment]
Good news,I just installed the latest github version of tmap and it works now. Very odd. My previous install of tmap was also from github and was at most a few days old.

For your info, when the error happened, I did not get the message about the stars downsampling. Also plot(bg) worked, so it was not an issue with OpenStreetMap/Java/etc but likely was somewhere between tmap and stars.

But anyway it works now. Thanks for this and many thanks for your great work on the tmap package as a whole! It is one of my very favorite rspatial packages!
